x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -822,6 +819,12 @@ export class Authenticator {
         searchParams.append(key, value);
       }
     }
+    if (!providerType) {
+      providerType =
+        this.options.config.authc.sortedProviders.length > 0
+          ? this.options.config.authc.sortedProviders[0].type
+          : '';
+    }
```

Comment:
__nit:__ having an empty string as a provider type feels a bit weird, how about just using `undefined` instead with something like this?
```ts
    providerType = providerType ?? this.options.config.authc.sortedProviders[0]?.type;
    ...
    return this.options.config.authc.selector.enabled ||
      (providerType && shouldProviderUseLoginForm(providerType))
    ...
```